### PR TITLE
Make the Github polling more resilient.

### PR DESCRIPTION
**Pull request by seanparsons**
**Problem:**
There were a couple of issues with the previous polling:
- It was being (re)triggered through changes in a React `useEffect` call, which could trigger a lot of `clearInterval` and `setInterval` calls as various dependencies of the call changed.
- As multiple parts of the polling were updating a `GithubData` value, they were all setting their own value for `lastUpdatedAt` which resulted in the timer in the UI bouncing around near 0s for a few seconds.
- The various promises triggered in the polling weren't `await`ed or anything like that so in the case of a slow call could dogpile the server.

**Fix:**
The primary part of improving the polling was to have all the various parts of the polling have their results be accumulated in the polled function, which had the following effect:
- With a tweak to the action, all the updates to `GithubData` are now happening in quick succession so the `lastUpdatedAt` value doesn't visibly get bounced about.
- Now they are all effectively being waited on to get their results, the polling function only succeeds once all of the promises have finished.

The triggering of the polling has been changed to live completely outside the sphere of React as well, with it scheduling the next run only once the previous poll job has finished.

**Commit Details:**
- Moved updating of `GithubData.lastUpdatedAt` to the action handler from the action creator so that it should be changing monotonically. As well as that being the exact time it does change.
- Removed React based polling from `editor-component.tsx` and added a new function for polling based on the time when the last poll completed, which now accesses the Zustand store more directly.
- New polling function `startGithubPolling` is triggered from the `Editor` component.
- Added `dispatchPromiseActions` utility function.
- Refactored a lot of the functions called during the polling to return the actions that they previously individually dispatched for the result of processing them. As they are now collated together and dispatched all together.
- Refactored out `updateUpstreamChanges` from the previous polling code.

### CODE REVIEW

File: editor/src/core/shared/github.ts [hunk 1]
```diff
@@ -355,66 +367,61 @@ export async function updatePullRequestsForBranch(
...
             ),
-          ],
-          'everyone',
-        )
-        break
-      case 'SUCCESS':
-        dispatch(
-          [
+          ]
+          break
```
Comment (ruggi) [inline editor/src/core/shared/github.ts:402]: not needed

File: editor/src/core/shared/github.ts [hunk 2]
```diff
@@ -355,66 +367,61 @@ export async function updatePullRequestsForBranch(
...
             }),
-          ],
-          'everyone',
-        )
-        break
-      default:
-        const _exhaustiveCheck: never = responseBody
-        throw new Error(`Unhandled response body ${JSON.stringify(responseBody)}`)
+          ]
+          break
```
Comment (ruggi) [inline editor/src/core/shared/github.ts:412]: not needed

**Review by ruggi — COMMENTED** [review]
LGTM! There's only one issue though, which is that now repos and branches don't refresh either on load or when they change. I'll push a small fix itmt

**Comment by ruggi** [comment]
> refresh repos/branch on load/change

this one → https://github.com/concrete-utopia/utopia/pull/2920/commits/aa435632a402cfe7a6a7c4915c33706bfdcb6ed6

**Review by ruggi — APPROVED** [review]

**Comment by seanparsons** [comment]
> > LGTM! There's only one issue though, which is that now repos and branches don't refresh either on load or when they change. I'll push a small fix itmt
> 
> this one → [aa43563](https://github.com/concrete-utopia/utopia/commit/aa435632a402cfe7a6a7c4915c33706bfdcb6ed6)

Good spot, thank you!

**Review by Rheeseyb — APPROVED** [review]

File: editor/src/core/shared/github.ts (hunk 1 above)
Comment (seanparsons) [reply editor/src/core/shared/github.ts]: Fixed in cc6fda6c71313001c3045ec5dd87329b0039f5b4.

File: editor/src/core/shared/github.ts (hunk 2 above)
Comment (seanparsons) [reply editor/src/core/shared/github.ts]: Fixed in cc6fda6c71313001c3045ec5dd87329b0039f5b4.

**Pull request merged**